File: lab1/fp_SimUDuck/RedheadDuck.cpp
#include "stdafx.h"
#include "RedheadDuck.h"

RedheadDuck::RedheadDuck()
	: Duck(FlyBehaviors::GetFlyWithWings(),
		QuackBehaviors::GetQuackBehavior(),
		DanceBehaviors::GetMinuetDanceBehavior())
{
}

void RedheadDuck::Display() const
{
	std::cout << "I'm redhead duck" << std::endl;
}

File: lw1_fn/src/duck/RedheadDuck.php
<?php

class RedheadDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            FlyBehavior\getFlyWithWings(),
            QuackBehavior\getQuackBehavior(),
            DanceBehavior\getMinuetDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m redhead duck!' . PHP_EOL;
    }
}